[sonic-head.yang]: Libyang sweep host bits from inet:ip-prefix, fixin…

praveen-li opened the pull request:
…g that problem.

Changed inet:ip-prefix to custom ip-prefix.
Changed test case accordingly.

<!--
Please make sure you've read and understood our contributing guidelines;
https://github.com/Azure/SONiC/blob/gh-pages/CONTRIBUTING.md

** Make sure all your commits include a signature generated with `git commit -s` **

If this is a bug fix, make sure your description includes "fixes #xxxx", or
"closes #xxxx"

Please provide the following information:
-->

**- What I did**
Libyang sweep host bits from inet:ip-prefix, fixing that problem.

**- How I did it**
    Changed inet:ip-prefix to custom ip-prefix.
    Changed test case accordingly.
**- How to verify it**

Build the package and also wrote custom test case to print config diff.



```
pchaudha@e8879b00ab3e:/sonic/src/sonic-yang-mgmt/tests/yang-model-tests/custom_code$ python pc_yang.py
Loaded below Yang Models
['sonic-vlan', 'sonic-head', 'sonic-portchannel', 'sonic-acl', 'sonic-loopback-interface', 'sonic-port', 'sonic-interface']
{u'INTERFACE': {insert: {'Ethernet112|2a04:5555:40:a709::/126': {u'scope': 'global', u'family': 'IPv6'}, 'Ethernet14|10.184.229.210/31': {u'scope': 'global', u'family': 'IPv4'}, 'Ethernet16|10.184.230.210/31': {u'scope': 'global', u'family': 'IPv4'}, 'Ethernet14|2a04:5555:40:a749::/126': {u'scope': 'global', u'family': 'IPv6'}, 'Ethernet18|10.184.231.210/31': {u'scope': 'global', u'family': 'IPv4'}, 'Ethernet16|2a04:5555:40:a789::/126': {u'scope': 'global', u'family': 'IPv6'}, 'Ethernet112|10.184.228.210/31': {u'scope': 'global', u'family': 'IPv4'}, 'Ethernet18|2a04:5555:40:a7c9::/126': {u'scope': 'global', u'family': 'IPv6'}}, delete: {u'Ethernet16|10.184.230.211/31': {u'scope': u'global', u'family': u'IPv4'}, u'Ethernet112|10.184.228.211/31': {u'scope': u'global', u'family': u'IPv4'}, u'Ethernet112|2a04:5555:40:a709::2/126': {u'scope': u'global', u'family': u'IPv6'}, u'Ethernet18|2a04:5555:40:a7c9::2/126': {u'scope': u'global', u'family': u'IPv6'}, u'Ethernet14|2a04:5555:40:a749::2/126': {u'scope': u'global', u'family': u'IPv6'}, u'Ethernet16|2a04:5555:40:a789::2/126': {u'scope': u'global', u'family': u'IPv6'}, u'Ethernet14|10.184.229.211/31': {u'scope': u'global', u'family': u'IPv4'}, u'Ethernet18|10.184.231.211/31': {u'scope': u'global', u'family': u'IPv4'}}}, u'VLAN_INTERFACE': {insert: {'Vlan100|2a04:5555:45:6709::/64': {u'scope': 'global', u'family': 'IPv6'}, 'Vlan777|fe80::/10': {u'scope': 'local', u'family': 'IPv6'}, 'Vlan777|2a04:5555:41:4e9::/64': {u'scope': 'global', u'family': 'IPv6'}, 'Vlan777|10.111.58.64/26': {u'scope': 'global', u'family': 'IPv4'}, 'Vlan100|fe80::/10': {u'scope': 'local', u'family': 'IPv6'}, 'Vlan100|10.222.10.64/26': {u'scope': 'global', u'family': 'IPv4'}}, delete: {u'Vlan777|fe80::1/10': {u'scope': u'local', u'family': u'IPv6'}, u'Vlan100|2a04:5555:45:6709::1/64': {u'scope': u'global', u'family': u'IPv6'}, u'Vlan777|10.111.58.65/26': {u'scope': u'global', u'family': u'IPv4'}, u'Vlan777|2a04:5555:41:4e9::1/64': {u'scope': u'global', u'family': u'IPv6'}, u'Vlan100|10.222.10.65/26': {u'scope': u'global', u'family': u'IPv4'}, u'Vlan100|fe80::1/10': {u'scope': u'local', u'family': u'IPv6'}}}}
Xlate and Rev Xlate Passed



pchaudha@e8879b00ab3e:/sonic/src/sonic-yang-mgmt/tests/yang-model-tests/custom_code$ python pc_yang.py
Loaded below Yang Models
['sonic-vlan', 'sonic-head', 'sonic-portchannel', 'sonic-acl', 'sonic-loopback-interface', 'sonic-port', 'sonic-interface']
Xlate and Rev Xlate Passed
```

```
pchaudha@e8879b00ab3e:/sonic$ make -f slave.mk target/python-wheels/sonic_yang_mgmt-1.0-py2-none-any.whl
SONiC Build System

Build Configuration
"CONFIGURED_PLATFORM"             : "broadcom"
"CONFIGURED_ARCH"                 : "amd64"
"SONIC_CONFIG_PRINT_DEPENDENCIES" : ""
"SONIC_BUILD_JOBS"                : "1"
"SONIC_CONFIG_MAKE_JOBS"          : "24"
"SONIC_USE_DOCKER_BUILDKIT"       : ""
"USERNAME"                        : "admin"
"PASSWORD"                        : "YourPaSsWoRd"
"ENABLE_DHCP_GRAPH_SERVICE"       : ""
"SHUTDOWN_BGP_ON_START"           : ""
"ENABLE_PFCWD_ON_START"           : ""
"INSTALL_DEBUG_TOOLS"             : ""
"ROUTING_STACK"                   : "frr"
"FRR_USER_UID"                    : "300"
"FRR_USER_GID"                    : "300"
"ENABLE_SYNCD_RPC"                : ""
"ENABLE_ORGANIZATION_EXTENSIONS"  : "y"
"HTTP_PROXY"                      : ""
"HTTPS_PROXY"                     : ""
"ENABLE_SYSTEM_TELEMETRY"         : "y"
"ENABLE_RESTAPI"                  : "n"
"ENABLE_ZTP"                      : ""
"SONIC_DEBUGGING_ON"              : ""
"SONIC_PROFILING_ON"              : ""
"KERNEL_PROCURE_METHOD"           : "build"
"BUILD_TIMESTAMP"                 : ""
"BLDENV"                          : ""
"VS_PREPARE_MEM"                  : "yes"
"ENABLE_SFLOW"                    : "y"

pchaudha@e8879b00ab3e:/sonic$
```

Testing for Commit 2:
```
Build Configuration
"CONFIGURED_PLATFORM"             : "broadcom"
"CONFIGURED_ARCH"                 : "amd64"
"SONIC_CONFIG_PRINT_DEPENDENCIES" : ""
"SONIC_BUILD_JOBS"                : "1"
"SONIC_CONFIG_MAKE_JOBS"          : "24"
"SONIC_USE_DOCKER_BUILDKIT"       : ""
"USERNAME"                        : "admin"
"PASSWORD"                        : "YourPaSsWoRd"
"ENABLE_DHCP_GRAPH_SERVICE"       : ""
"SHUTDOWN_BGP_ON_START"           : ""
"ENABLE_PFCWD_ON_START"           : ""
"INSTALL_DEBUG_TOOLS"             : ""
"ROUTING_STACK"                   : "frr"
"FRR_USER_UID"                    : "300"
"FRR_USER_GID"                    : "300"
"ENABLE_SYNCD_RPC"                : ""
"ENABLE_ORGANIZATION_EXTENSIONS"  : "y"
"HTTP_PROXY"                      : ""
"HTTPS_PROXY"                     : ""
"ENABLE_SYSTEM_TELEMETRY"         : "y"
"ENABLE_RESTAPI"                  : "n"
"ENABLE_ZTP"                      : ""
"SONIC_DEBUGGING_ON"              : ""
"SONIC_PROFILING_ON"              : ""
"KERNEL_PROCURE_METHOD"           : "build"
"BUILD_TIMESTAMP"                 : ""
"BLDENV"                          : ""
"VS_PREPARE_MEM"                  : "yes"
"ENABLE_SFLOW"                    : "y"

pchaudha@e8879b00ab3e:/sonic$

pchaudha@e8879b00ab3e:/sonic$ date
Wed Mar  4 07:53:38 UTC 2020


pchaudha@e8879b00ab3e:/sonic$ pip install target/python-wheels/sonic_yang_mgmt-1.0-py2-none-any.whl
Processing ./target/python-wheels/sonic_yang_mgmt-1.0-py2-none-any.whl
Installing collected packages: sonic-yang-mgmt
Successfully installed sonic-yang-mgmt-1.0
pchaudha@e8879b00ab3e:/sonic$ ls -l /var/pchaudha/.local/yang-models/
total 36
-rw-r--r-- 1 pchaudha gpchaudha 4959 Mar  4 07:52 sonic-acl.yang
-rw-r--r-- 1 pchaudha gpchaudha 2214 Mar  4 07:52 sonic-head.yang
-rw-r--r-- 1 pchaudha gpchaudha 2207 Mar  4 07:52 sonic-interface.yang
-rw-r--r-- 1 pchaudha gpchaudha 1379 Mar  4 07:52 sonic-port.yang
-rw-r--r-- 1 pchaudha gpchaudha 1440 Mar  4 07:52 sonic-portchannel.yang
-rw-r--r-- 1 pchaudha gpchaudha 3682 Mar  4 07:52 sonic-vlan.yang
-rw-r--r-- 1 pchaudha gpchaudha 5391 Mar  4 07:52 sonic_yang_tree
```

**- Description for the changelog**
<!--
Write a short (one line) summary that describes the changes in this
pull request for inclusion in the changelog:
-->


**- A picture of a cute animal (not mandatory but encouraged)**

Commit:
[sonic-head.yang]: Libyang sweep host bits from inet:ip-prefix, fixing that problem.

Changed inet:ip-prefix to custom ip-prefix.
Changed test case accordingly.

zhenggen-xu commented on src/sonic-yang-mgmt/yang-models/Sonic Yang Tree:
```diff
@@ -52,22 +52,37 @@ module: sonic-acl
 module: sonic-interface
```
Is file autogenerated?  If so, should we not keep this as part of the code?

praveen-li replied:
This file is generated by Pyang tool. Yes, Cool, we can also generate it as part of build process. Sonic-stretch docker will\should have pyang installed and then while building sonic-yang-models pkg I can run "pyang" tool.

zhenggen-xu commented on src/sonic-yang-mgmt/yang-models/sonic-head.yang:
```diff
@@ -8,18 +8,40 @@ module sonic-head {
     description "Head yang Module for SONiC OS";
-    
+
     revision 2019-07-01 {
```
What is the significant of this revision?  Do we need make changes here after the changes in this PR?

praveen-li replied:
Ideally each PR in github master should have new revision number. But since it is not merged in MSFT github yet. I kept same revision number for all PR till first merge.

zhenggen-xu commented:
Should we do the same for loopback interface too?

zhenggen-xu commented on src/sonic-yang-mgmt/yang-models/sonic-vlan.yang:
```diff
@@ -75,8 +75,10 @@ module sonic-vlan {
 				}
 
 				leaf ip-prefix {
-					mandatory true;
```
So this is not needed or we had too restricted before?

praveen-li replied:
Yang List Keys are mandotary by default. So this statement was redundant.

praveen-li commented:
> Should we do the same for loopback interface too?

Yes, I missed loopback interface, I need to fix it for loopback as well.

Commit:
[setup.py]: Build yang tree while package build.

Changes done:
1.) Build yang tree while package build.
2.) Add sonic_yang_tree as part of package.
3.) Fix sonic-loopback-interface.yang for ip-prefix problem.
